redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseAppContent.kt

```diff
@@ -388,35 +389,49 @@ private class ViewContentCodeBinding<A : AppService>(
       treehouseUiOrNull = treehouseUi
       val restoredId = viewOrNull?.stateSnapshotId
       val restoredState = if (restoredId != null) app.stateStore.get(restoredId.value.orEmpty()) else null
-      treehouseUi.start(
-        changesSink = this@ViewContentCodeBinding,
-        onBackPressedDispatcher = object : OnBackPressedDispatcherService {
-          override fun addCallback(onBackPressedCallback: OnBackPressedCallbackService): CancellableService {
-            app.dispatchers.checkZipline()
-            val cancellable = onBackPressedDispatcher.addCallback(
-              object : OnBackPressedCallback(onBackPressedCallback.isEnabled) {
-                override fun handleOnBackPressed() {
-                  bindingScope.launch(app.dispatchers.zipline) {
-                    onBackPressedCallback.handleOnBackPressed()
-                  }
-                }
-              },
-            )
-            return object : CancellableService {
-              override fun cancel() {
-                app.dispatchers.checkZipline()
-                cancellable.cancel()
-              }
-
-              override fun close() {
-                cancel()
-              }
+      try {
+        treehouseUi.start(
+          changesSink = this@ViewContentCodeBinding,
+          onBackPressedDispatcher = onBackPressedDispatcherService,
+          uiConfigurations = uiConfigurationFlow,
+          stateSnapshot = restoredState,
+        )
+      } catch (e: ZiplineApiMismatchException) {
```

Comment:
This is the new thing

Reply:
How should we keep track of when we can delete things like this?

Reply:
Very good question. How about we borrow an idea from Renovate and do a GitHub tracking issue with all of our in-flight deprecations?
https://github.com/cashapp/redwood/issues/1539